Revamp Demo/Catalog

liquidx opened the pull request:
All demo pages use mdc-toolbar.
All demo pages use black/green theme from Material Components.
All demo pages have a hero section that shows the representative example of component.
All demo pages use Roboto Mono for the Toolbar font.

Adds demos.css to be included in all demos to share style.
Adds a CSS reset in demos.css to clear up styles in the demo.

liquidx commented:
It's pretty much ready for review. The CSS is not the best because there's not a lot of consistency between each component demo, but this consolidates the styling some what.

Let me know if there's a demo template that I should update with the new structure.

Staging version of this is here:
https://staging-dot-material-components-web.appspot.com/

appsforartists commented:
Seeing a nasty [FOUC](https://en.wikipedia.org/wiki/Flash_of_unstyled_content) on every click on my Chromebook.

liquidx commented:
1. I'm not sure about what is going on with the Ripple. Since it may mean touching the component, maybe that should be done in a separate CL. I've been trying to only touch the demo/catalog in this PR. I can file an issue for that.

2. Good point about the "vanilla" theme being hidden. On the flip side, this does ensure we actually support the theming on all our components. I think the consistency with the look of our docs site is important, but I would support adding example where possible of what the default component would look like.

**Card**
Fixed both issues.

**Checkbox**
Fixed and changed highlight to green.

**Dialog**
Fixed.

** Grid List **
Fixed, also replaced the Toggle RTL button with a checkbox.

** Lists **
Ah yes, the list groups look weird with the borders. I'll fix them.

** Select **
Fixed fully featured demo. Oops, used the same id in two places.

**Simple Menu**
Fixed issues with flexbox.

**Snackbar**
Sorry, I think my reordering of the CSS rules exposed that we didn't override the snackbar transform offscreen during loading. Fixed now.

**Toolbar**
Fixed back button issue.
I deliberately left the last two examples as is since it was demonstrating some color changes.

**Typography**
Fixed by adding an override to this.

liquidx commented:
As for the FOUC - this was already happening before I started styling the demos.

I suggest doing a sweep over all the demos to ensure we hide some of the visibility until page loaded. I had to do some of that in the snackbar demo.

liquidx commented on demos/card.html:
```diff
@@ -134,123 +121,207 @@ <h2 class="mdc-card__subtitle">Subtitle here</h2>
-      <div class="mdc-card demo-card">
-        <div class="mdc-card__horizontal-block">
+      <div>
+        <style>
```
Done. Not sure why they were here in the first place.

liquidx commented on demos/card.html:
```diff
@@ -134,123 +121,207 @@ <h2 class="mdc-card__subtitle">Subtitle here</h2>
-            <button class="mdc-button mdc-button--compact mdc-card__action">A2</button>
+      <div>
+        <style>
+          .demo-card--small .mdc-card__media {
```
Done, as above, not sure why they were here in the first place.

liquidx commented on demos/demos.css:
```diff
@@ -0,0 +1,111 @@
+
```
Done

liquidx replied:
Fixed

liquidx commented on demos/demos.css:
```diff
@@ -0,0 +1,111 @@
+
+/* http://meyerweb.com/eric/tools/css/reset/
```
Done

liquidx commented on demos/demos.css:
```diff
@@ -0,0 +1,111 @@
+}
+
+body {
+  font-family: Roboto, -apple-system, Arial, sans-serif;
```
-apple-system will pull in San Francisco. But agreed, let's pull that out for now.

liquidx commented on demos/demos.css:
```diff
@@ -0,0 +1,111 @@
+  font-family: Roboto, -apple-system, Arial, sans-serif;
+}
+
+p {
```
Indeed, removed.

liquidx commented on demos/fab.html:
```diff
@@ -39,12 +43,57 @@
+
+      /* Override colors to match catalog theme. */
+
+      .mdc-fab {
```
Good point, removed.

liquidx commented on demos/fab.html:
```diff
@@ -39,12 +43,57 @@
+        background-color: #212121;
+      }
+
+      .mdc-fab--plain {
```
Removed, may had been something I had to do when I had the CSS imports in the wrong order.

liquidx commented on demos/fab.html:
```diff
@@ -39,12 +43,57 @@
+
+      /* Override mini-FABs with smaller icon size. */
+      .mdc-fab--mini.material-icons .mdc-fab__icon {
+        font-size: 18px;
```
The current icon is too big though for the mini fab. I'll take it up with the design team and remove it.

liquidx commented on demos/list.html:
```diff
@@ -60,32 +62,139 @@
+        <aside>
+          <p>
+          <em>NOTE:</em> For the purposes of this demo, we've set a max-width of 600px on all
+          <code>mdc-list</code> elements surrounded by a 1px border. This is
```
Done

liquidx commented on demos/select.html:
```diff
@@ -122,13 +173,15 @@
       </section>
 
-      <section>
+      <!--
```
Done

liquidx commented on demos/simple-menu.html:
```diff
@@ -17,11 +17,13 @@
 <html>
```
I made the menu shorter and the box taller so it accomodated it in more screen sizes.

appsforartists commented:
Just took a quick skim - looks rad Alastair!

I wonder how much of the list should be clickable.  If it was mobile, the whole row would be a link.  Is that too wide for a larger screen?

At the very least, I think the icons should be clickable.

liquidx commented on demos/grid-list.html:
```diff
@@ -27,637 +29,720 @@
-        var gridListEl = elem.parentElement.querySelector('#example');
-        gridListEl.getAttribute('dir') === 'rtl' ? gridListEl.setAttribute('dir', 'ltr') : gridListEl.setAttribute('dir', 'rtl');
-      }
+      // function toggleRTL(elem) {
```
Done

liquidx commented on demos/grid-list.html:
```diff
@@ -27,637 +29,720 @@
           mdc.gridList.MDCGridList.attachTo(node);
         }
       }
+      // Hook up RTL
```
Done

liquidx commented on demos/simple-menu.html:
```diff
@@ -32,10 +34,23 @@
       }
 
       .demo-content {
-        position: relative;
```
Done

liquidx commented:
Right on! My first commit :)
